cryptography/src/lib.rs

```diff
@@ -124,6 +124,42 @@ pub trait Hasher: Clone + Send + Sync + 'static {
     fn random<R: Rng + CryptoRng>(rng: &mut R) -> Digest;
 }
 
+pub trait BatchScheme {
+    /// Create a new Batch
+    fn new() -> Self;
+
+    /// Append item to the batch
+    ///
+    /// The message should not be hashed prior to calling this function. If a particular scheme
+    /// requires a payload to be hashed before it is signed, it will be done internally.
+    ///
+    /// A namespace should be used to prevent replay attacks. It will be prepended to the message so
+    /// that a signature meant for one context cannot be used unexpectedly in another (i.e. signing
+    /// a message on the network layer can't accidentally spend funds on the execution layer). See
+    /// [union_unique](commonware_utils::union_unique) for details
+    fn add(
+        &mut self,
+        namespace: Option<&[u8]>,
+        message: &[u8],
+        public_key: &PublicKey,
+        signature: &Signature,
+    ) -> Result<(), BatchError>;
+
+    /// Perform batch verification over the previously added items.
+    fn verify<R: RngCore + CryptoRng>(self, rng: R) -> bool;
+}
+
+#[derive(Debug, PartialEq)]
+pub enum BatchError {
```

Comment:
added a Basic `BatchError` `InvalidItem` but it could be splitted into two BatchError:
- InvalidPublicKey
- InvalidSignature

Reply:
Like in `Scheme`, let's just use `bool` instead of defining a custom error type.

Worth a discussion of if this is the right approach, however, I find it to be a useful way to avoid going insane when each library has its own approach to error-handling.

Reply:
I agree, moreover having just `one error` as I did without any details does not give more information than having a `bool`